packet_test.go

```diff
@@ -1129,3 +1129,256 @@ func BenchmarkMarshalTo(b *testing.B) {
+					0x98, 0x36, 0xbe, 0x88, 0x9e,
+				},
+				Raw: []byte{
+					0x90, 0xe0, 0x69, 0x8f, 0xd9, 0xc2, 0x93, 0xda, 0x1c, 0x64,
+					0x27, 0x82, 0x10, 0x00, 0x00, 0x07, 0x05, 0x18, 0xAA, 0xAA,
+					0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA,
+					0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA, 0xAA,
+					0xAA, 0xAA, 0x00, 0x00, 0x98, 0x36, 0xbe, 0x88, 0x9e,
+				},
+			},
+		},
+		"PacketMultipleTwoByteExtensions": {
+			rawPkt: []byte{
+				0x90, 0xe0, 0x69, 0x8f, 0xd9, 0xc2, 0x93, 0xda, 0x1c, 0x64,
+				0x27, 0x82, 0x10, 0x00, 0x00, 0x03, 0x01, 0x00, 0x02, 0x01,
+				0xBB, 0x00, 0x03, 0x04, 0xCC, 0xCC, 0xCC, 0xCC, 0x98, 0x36,
+				0xbe, 0x88, 0x9e,
+			},
+			expectedPacket: &Packet{Header: Header{
+				Marker:           true,
+				Extension:        true,
+				ExtensionProfile: 0x1000,
+				Extensions: []Extension{
+					{1, []byte{}},
+					{2, []byte{
+						0xBB,
+					}},
+					{3, []byte{
+						0xCC, 0xCC, 0xCC, 0xCC,
+					}},
+				},
+				Version:        2,
+				PayloadType:    96,
+				PayloadOffset:  28,
+				SequenceNumber: 27023,
+				Timestamp:      3653407706,
+				SSRC:           476325762,
+				CSRC:           []uint32{},
+			},
+				Payload: []byte{
+					0x98, 0x36, 0xbe, 0x88, 0x9e,
+				},
+				Raw: []byte{
+					0x90, 0xe0, 0x69, 0x8f, 0xd9, 0xc2, 0x93, 0xda, 0x1c, 0x64,
+					0x27, 0x82, 0x10, 0x00, 0x00, 0x03, 0x01, 0x00, 0x02, 0x01,
+					0xBB, 0x00, 0x03, 0x04, 0xCC, 0xCC, 0xCC, 0xCC, 0x98, 0x36,
+					0xbe, 0x88, 0x9e,
+				},
+			},
+		},
+		"DefaultExtension": {
+			rawPkt: []byte{
+				0x80, 0xe0, 0x69, 0x8f, 0xd9, 0xc2, 0x93, 0xda, 0x1c, 0x64,
+				0x27, 0x82, 0x00, 0x01, 0x98, 0x36, 0xbe, 0x88, 0x9e,
+			},
+			expectedPacket: &Packet{Header: Header{
+				Marker:           true,
+				Extension:        false,
+				ExtensionProfile: 0,
+				Version:          2,
+				PayloadOffset:    12,
+				PayloadType:      96,
+				SequenceNumber:   27023,
+				Timestamp:        3653407706,
+				SSRC:             476325762,
+				CSRC:             []uint32{},
+			},
+				Payload: []byte{
+					0x00, 0x01, 0x98, 0x36, 0xbe, 0x88, 0x9e,
+				},
+				Raw: []byte{
+					0x80, 0xe0, 0x69, 0x8f, 0xd9, 0xc2, 0x93, 0xda, 0x1c, 0x64,
+					0x27, 0x82, 0x00, 0x01, 0x98, 0x36, 0xbe, 0x88, 0x9e,
+				},
+			},
+		},
+	}
+
+	for name, testCase := range cases {
+		t.Run(name, func(t *testing.T) {
+			p := &Packet{}
+			// First Unmarshal
+			if err := p.Unmarshal(testCase.rawPkt); err != nil {
+				t.Error("Test multiple unmarshal: first: %w", err)
+			} else if len(p.Extensions) != len(testCase.expectedPacket.Extensions) {
+				t.Errorf("Test multiple unmarshal: first: got %d extensions, want %d", len(p.Extensions), len(testCase.expectedPacket.Extensions))
+			} else if !reflect.DeepEqual(p, testCase.expectedPacket) {
+				t.Errorf("TestMultiple unmarshal: first: got %#v, want %#v", p, testCase.expectedPacket)
+			}
+
+			// Second Unmarshal
+			if err := p.Unmarshal(testCase.rawPkt); err != nil {
```

Comment:
Oh, I see.  You see if the same packet being unmarshalled twice produces oddness.